src/custom/auth.ts

```diff
@@ -1,60 +1,134 @@
 import { InfisicalSDK } from "..";
 import { ApiV1AuthUniversalAuthLoginPostRequest } from "../infisicalapi_client";
 import { DefaultApi as InfisicalApi } from "../infisicalapi_client";
+import { AuthMethod, TAuthCredentials, TTokenDetails } from "../types";
 import { MACHINE_IDENTITY_ID_ENV_NAME } from "./constants";
+import { InfisicalError, newInfisicalError } from "./errors";
 import { getAwsRegion, performAwsIamLogin } from "./util";
 
-type AuthenticatorFunction = (accessToken: string) => InfisicalSDK;
+type TAuthenticator = {
+	authenticate: (tokenDetails: Omit<TTokenDetails, "fetchedTime" | "firstFetchTime">) => Promise<InfisicalSDK>;
+	setCredentials: (credentials: TAuthCredentials) => void;
+};
 
 type AwsAuthLoginOptions = {
 	identityId?: string;
 };
 
+export const renewToken = async (apiClient: InfisicalApi, token?: string) => {
```

Comment:
Suggestion: Add  JSDoc comment to explain the purpose, parameters, return type, and possible errors of the function. This is helpful for maintainability and for other developers who might use this function.
'''
/**
 * Renews the access token using the provided Infisical API client.
 *
 * @param apiClient - An instance of InfisicalApi to make the request.
 * @param token - The current access token to be renewed.
 * @returns The renewed token data.
 * @throws {InfisicalError} If the token is missing or the renewal request fails.
 */
'''